docs/source/tutorials/token_classification/tutorial-p2.ipynb

```diff
@@ -0,0 +1,657 @@
+    "## 1. Install the required dependencies "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "a1349304",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from termcolor import colored \n",
+    "from cleanlab.filter import find_label_issues \n",
+    "from rank import get_label_quality_scores \n",
+    "from utils import * \n",
+    "# change to this after token_classification.rank is pushed to the official package \n",
+    "# from cleanlab.token_classification.rank import get_label_quality_scores "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9ad75b45",
+   "metadata": {},
+   "source": [
+    "## 2. Get `pred_probs` and `labels` \n",
+    "\n",
+    "For more information on how to get `pred_probs` and `labels`, see part 1. Recall that we also need the number of word-level tokens for each sentence. Also get their nested list format. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "519cb80c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "labels = dict(np.load('labels.npz')) \n",
+    "pred_probs = dict(np.load('pred_probs.npz')) \n",
+    "\n",
+    "labels_nl = to_nl(labels) \n",
+    "pred_probs_nl = to_nl(pred_probs) \n",
+    "\n",
+    "labels = [label for labels in labels_nl for label in labels] \n",
+    "pred_probs = np.array([pred_prob for pred_probs in pred_probs_nl for pred_prob in pred_probs]) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1dc3150f",
+   "metadata": {},
+   "source": [
+    "## 3. Use cleanlab to find potential label issues \n",
+    "\n",
+    "Based on the given labels and out-of-sample predicted probabilities, cleanlab can quickly help us identify label issues in our dataset. Here we request that the indices of the identified label issues be sorted by cleanlab’s self-confidence score, which measures the quality of each given label via the probability assigned to it in our model’s prediction."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "95dc7268",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Cleanlab found 611 potential label issues. \n",
+      "The top 10 most likely label errors:\n",
+      "[32203 31970 40344 34547  6727 31531 34553 46388 28663 13979]\n"
+     ]
+    }
+   ],
+   "source": [
+    "issues = find_label_issues(labels, pred_probs, return_indices_ranked_by='self_confidence') \n",
+    "top = 10 \n",
+    "print('Cleanlab found %d potential label issues. ' % len(issues)) \n",
+    "print('The top %d most likely label errors:' % top) \n",
+    "print(str(issues[:top])) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1e6e2963",
+   "metadata": {},
+   "source": [
+    "Note that the indices are in the flattened format. We map them to a tuple `(i, j)`, which corresponds to the `j`'th word-level token of the `i`'th sentence. "
```

Comment:
As discussed let's not use the flattened format, and never have the user need to worry about flattened format.

Reply:
`pred_probs` and `labels` still need to be flattened to feed into `find_label_issues`, so we need to build a `index -> (i, j)` mapping. Not sure if there's a more elegant way to resolve this

Reply:
Didn't we talk about the user never having to flatten those? 

I really think we should not require the user to flatten those. Why can't our `cleanlab.token_classification.filter.find_label_issues` function simply take in unflattened `pred_probs` & `labels` and then do the flattening/unflattening as necessary internally inside of this function.

Reply:
Right, will fix this